aws-api/src/test/java/com/amplifyframework/api/aws/GsonGraphQLResponseFactoryTest.java

```diff
@@ -122,15 +123,59 @@ public void partialResponseRendersWithTodoDataAndErrors() throws ApiException {
         // Assert that we parsed the errors successfully.
         assertNotNull(response.getErrors());
 
-        final List<GraphQLResponse.Error> expectedErrors = Arrays.asList(
-            new GraphQLResponse.Error("failed"),
-            new GraphQLResponse.Error("failed"),
-            new GraphQLResponse.Error("failed")
-        );
+        for (GraphQLResponse.Error error : response.getErrors()) {
+            assertEquals(error.getMessage(), "failed");
+        }
+    }
 
-        final List<GraphQLResponse.Error> actualErrors = response.getErrors();
+    /**
+     * This tests the GsonErrorDeserializer.  The test JSON response has 4 errors, which are all in
+     * different formats, but are expected to be parsed into the same resulting object:
+     * 1. Error contains errorType, errorInfo, data (AppSync specific fields) at root level
+     * 2. Error contains errorType, errorInfo, data inside extensions object
+     * 3. Error contains errorType, errorInfo, data at root AND inside extensions (fields inside
+     * extensions take precedence)
+     * 4. Error contains errorType at root, and errorInfo, data inside extensions (all should be
+     * merged into extensions)
+     *
+     * @throws ApiException From API configuration
+     */
+    @Test
+    public void errorResponseDeserializesExtensionsMap() throws ApiException {
+        // Arrange some JSON string from a "server"
+        final String partialResponseJson =
+                Resources.readAsString("error-extensions-gql-response.json");
+
+        // Act! Parse it into a model.
+        final GraphQLResponse<ListTodosResult> response =
+                responseFactory.buildSingleItemResponse(partialResponseJson, ListTodosResult.class);
 
-        assertEquals(expectedErrors, actualErrors);
+        // Assert that the response contained things...
+        assertNotNull(response);
+        assertNotNull(response.getErrors());
+        assertEquals(response.getErrors().size(), 4);
+
+        // Assert that each error is has been parsed to the same output object
```

Comment:
Under the hood, [`assertEquals(..., ...)`](https://github.com/junit-team/junit4/blob/master/src/main/java/org/junit/Assert.java#L99-L122) is basically just this:
```
if (!expect.equals(actual)) {
    throw new AssertionError(providedMessage);
}
```

So, so long as your objects have a sensible `equals()` implementation (they do!), you can make high-level assertions on objects using jUnit's `assertEquals(...)` methods.

Here, what I propose is to reword this into one bit assertion statement, like:
```
assertEquals(
    /* expected */ new GraphQLResponse(
        null,
        Arrays.asList(
            new GraphQLError("Confict resolver rejects ...", ,, ) // build these up, from the fields in your assertions
        )
    ),
    actualResponse /* built the same exact way you are now, in the Act portion. */
);
```

Reply:
Originally I didn't do this because `GraphQLLocation` (which is only created via gson) didn't have a constructor, but I added that constructor while addressing @raphkim's feedback, essentially because a customer may actually want a constructor.  

I do like the suggestion generally because it is a good way to ensure everything is tested, even if new fields are added later, so I've gone ahead and made this change.  I did keep a couple intermediate asserts though.  When the test succeeds, they are redundant, but when the test fails, the intermediate ones are very useful to debug the problem.

I even caught and fixed a bug in after refactoring this test, which was that elements equal to JsonNull in the error object were getting dropped instead of getting merged into extensions.

Note: I did have to implement `equals` in the GraphQLLocation, GraphQLPathSegment, AppSyncExtensions classes to get this to work, but it's good that it's there now :)